### PR TITLE
Add tests for status propagation from Application to Source

### PR DESCRIPTION
**Pull request by lpichler**
Added tests for https://github.com/RedHatInsights/sources-api-go/pull/294

### CODE REVIEW

File: application_handlers_test.go [hunk 1]
```diff
@@ -618,6 +658,30 @@ func TestApplicationEdit(t *testing.T) {
...
+		t.Errorf("Invalid email notification data:")
+	}
+
+	src, err = sourceDao.GetById(&application.SourceID)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if src.AvailabilityStatus == previousSourceStatus {
+		t.Errorf("Invalid email notification data:")
```
Comment (MikelAlejoBR) [inline application_handlers_test.go]: I think you forgot the colons, ":", from the error messages

**Review by MikelAlejoBR — APPROVED** [review]
LGTM!

File: application_handlers_test.go [hunk 2]
```diff
@@ -618,6 +658,30 @@ func TestApplicationEdit(t *testing.T) {
...
+	if application.AvailabilityStatus == previousApplicationStatus {
+		t.Errorf("Invalid email notification data:")
+	}
+
+	src, err = sourceDao.GetById(&application.SourceID)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if src.AvailabilityStatus == previousSourceStatus {
```
Comment (petracihalova) [inline application_handlers_test.go:848]: in this condition, the `src.AvailabilityStatus` has value "available" and `previousSourceStatus` has value "", so we check that `"available" == "" `... but in this case we know that expected is value "available" ... from my perspective it is maybe better to have here something like ... what you think?

`if src.AvailabilityStatus != "available" {`

**Review by petracihalova — APPROVED** [review]
LGTM, I have only tiny comment

File: application_handlers_test.go (hunk 1 above)
Comment (lindgrenj6) [reply application_handlers_test.go]: @lpichler If you fix this message I'll merge this.

File: application_handlers_test.go (hunk 2 above)
Comment (lpichler) [reply application_handlers_test.go:848]: yes make sense but here we are testing whether status was changed and this condition cover only destination value after this update operation - I would leave as it is if you are ok with it anyway good insight

File: application_handlers_test.go (hunk 1 above)
Comment (lpichler) [reply application_handlers_test.go]: @lindgrenj6 message was fixed and also I changed words.

**Pull request merged**